File: one_and_two/one.cpp
#include <iostream>
#include <cstdio>
#include <cstdlib>

using namespace std;

int main()
{
    int case_num;
    scanf("%d", &case_num);
    for (int i = 0; i < case_num; i++)
    {
        int n;
        scanf("%d", &n);
        int num_twos = 0;
        int *arr = (int*) malloc(sizeof(int) * n);
        for (int j = 0; j < n; j++)
        {
            scanf("%d", arr + j);
            if (arr[j] == 2)
            {
                num_twos++;
            }
        }

        int k;
        int nums = 0;
        if (num_twos == 0)
        {
            printf("%d\n", 1);
        }

        else if (num_twos % 2 == 1)
        {
            printf("%d\n", -1);
        }

        else
        {
            for (int j = 0; nums < num_twos / 2; j++)
            {
                if (arr[j] == 2)
                {
                    k = j + 1;
                    nums++;
                }
            }

            printf("%d\n", k);
        }

        free(arr);
    }
}

File: one_and_two/one.py
def main():
    case_num = int(input())
    for _ in range(case_num):
        n = int(input())
        arr = input().split()
        arrs = [eval(i) for i in arr]
        num_twos = 0
        for num in arrs:
            if num == 2:
                num_twos += 1
                
        if (num_twos % 2 == 1):
            print(-1)
        
        elif (num_twos == 0):
            print(1)
            
        else:
            nums = 0
            j = 0
            while (nums < num_twos / 2):
                if (arrs[j] == 2):
                    k = j + 1
                    nums += 1
                j += 1
            print(k)
    
    
if __name__ == "__main__":
    main()